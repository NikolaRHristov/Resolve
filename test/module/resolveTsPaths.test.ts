import { rmSync, mkdirSync, copyFileSync, readFileSync } from "fs";
import { describe, expect, beforeEach, it } from "vitest";

import { resolveTsPaths } from "~/index";

describe(resolveTsPaths.name, () => {
  const files = [
    "root.d.ts",
    "root.js",
    "folder1/inside1.d.ts",
    "folder1/inside1.js",
    "folder1/folder2/inside2.d.ts",
    "folder1/folder2/inside2.js",
  ];

  beforeEach(() => {
    rmSync("test/fixtures/module/dist", { recursive: true, force: true });
    mkdirSync("test/fixtures/module/dist/folder1/folder2", { recursive: true });
    files.forEach((path) =>
      copyFileSync(
        `test/fixtures/module/src/${path}`,
        `test/fixtures/module/dist/${path}`
      )
    );
  });

  it("runs as a module", () => {
    resolveTsPaths({
      src: "test/fixtures/module/src",
      project: "test/fixtures/module/tsconfig.test.json",
    });
    files.forEach((file) =>
      expect(
        readFileSync(`test/fixtures/module/dist/${file}`).toString()
      ).toMatchSnapshot()
    );
  });
});
