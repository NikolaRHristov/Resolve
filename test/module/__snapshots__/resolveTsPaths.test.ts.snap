// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveTsPaths runs as a module 1`] = `
"export {};
"
`;

exports[`resolveTsPaths runs as a module 2`] = `
"export {};
"
`;

exports[`resolveTsPaths runs as a module 3`] = `
"import {} from \\"package\\";
import {} from \\"../root\\";
import {} from \\"~/non-existent-path\\";
"
`;

exports[`resolveTsPaths runs as a module 4`] = `
"import {} from \\"package\\";
import {} from \\"../root\\";
import {} from \\"~/non-existent-path\\";
"
`;

exports[`resolveTsPaths runs as a module 5`] = `
"import {} from \\"package\\";
import {} from \\"../../root\\";
import {} from \\"~/non-existent-path\\";
"
`;

exports[`resolveTsPaths runs as a module 6`] = `
"import {} from \\"package\\";
import {} from \\"../../root\\";
import {} from \\"~/non-existent-path\\";
"
`;
